src/pilot/discover_nodes/discover_nodes/dracclient/client.py

```diff
@@ -116,124 +116,7 @@ def set_idrac_settings(self, settings, idrac_fqdd='iDRAC.Embedded.1'):
-                LOG.debug("The iDRAC is pingable")
-
-            sleep(10)
-
-        if retries == 0 and ping_fail_count < required_ping_fail_count:
-            raise exceptions.DRACOperationFailed(drac_messages="Timed out "
-                                                 "waiting for the " +
-                                                 self.client.host +
-                                                 " iDRAC to "
-                                                 "become not pingable")
-
-        LOG.info("The iDRAC has become not pingable")
-
-        LOG.info("Waiting for the iDRAC to become pingable")
-        retries = 24
-        ping_success_count = 0
-        while retries > 0:
-            response = os.system("ping -c 1 {} 2>&1 1>/dev/null".format(
-                self.client.host))
-            retries -= 1
-            if response != 0:
-                LOG.debug("The iDRAC is not pingable")
-                ping_success_count = 0
-            else:
-                ping_success_count += 1
-                LOG.debug("The iDRAC is pingable, ping_success_count="
-                          "{}".format(ping_success_count))
-                if ping_success_count == 3:
-                    LOG.debug("Breaking")
-                    break
-
-            sleep(10)
-
-        if retries == 0 and ping_success_count < 3:
-            raise exceptions.DRACOperationFailed(drac_messages="Timed out "
-                                                 "waiting for the " +
-                                                 self.client.host +
-                                                 " iDRAC to "
-                                                 "become pingable")
-
-        LOG.info("The iDRAC has become pingable")
-        sleep(30)
-
-        LOG.info("Waiting for the iDRAC to become ready")
-        retries = 24
-        while retries > 0:
-            try:
-                is_ready = self.is_idrac_ready()
-                if is_ready:
-                    LOG.info("The iDRAC is ready")
-                    break
-                else:
-                    LOG.debug("The iDRAC is not ready")
-            except:
-                # It is normal to get a series of connection errors before
-                # the iDRAC becomes ready
-                ex = sys.exc_info()[0]
-                LOG.debug("An exception occurred while checking iDRAC ready "
-                          "state. Ignoring.: {}".format(str(ex)))
-                pass
-            retries -= 1
-            sleep(10)
-
-        if retries == 0:
-            raise exceptions.DRACOperationFailed(drac_messages="Timed out "
-                                                 "waiting for the " +
-                                                 self.client.host +
-                                                 " iDRAC to "
-                                                 "become ready")
+        return self._idrac_con.set_idrac_settings(idrac_fqdd, settings)
```

Comment:
Please see my comment above about the `_idrac_con` data member. If that's removed, this method, `set_idrac_settings()`, must be removed, too.

Reply:
Done, Please review